File: Chapter-5-Classes-and-Objects/009-a-sample-program-to-introduce-returning-objects.cpp
// a sample program to introduce returning objects
#include <bits/stdc++.h>

using namespace std;

class complexNumber /// x + yj form
{
    float x; /// real part
    float y; /// imaginary part
public:
    void input(float real, float imaginary)
    {
        x = real;
        y = imaginary;
    }
    friend complexNumber sum(complexNumber, complexNumber);
    void showComplexNumber(complexNumber c);
};

complexNumber sum(complexNumber one, complexNumber two)
{
    complexNumber add;
    add.x = one.x + two.x;
    add.y = one.y + two.y;
    return add;
}

void complexNumber::showComplexNumber(complexNumber c)
{
    cout << c.x << " + " << c.y << "j" << endl;
}

int main()
{
    complexNumber A, B, C;
    A.input(3.1, 5.65);
    B.input(2.75, 1.2);

    C = sum(A, B);
    cout << "A = ";
    A.showComplexNumber(A);
    cout << "B = ";
    B.showComplexNumber(B);
    cout << "C = ";
    C.showComplexNumber(C);
}

// output
// A = 3.1 + 5.65j
// B = 2.75 + 1.2j
// C = 5.85 + 6.85j

File: Chapter-5-Classes-and-Objects/009-a-sample-program-to-introduce-returning-objects.py
class ComplexNumber:
    def __init__(self):
        self.x = 0.0 # real part
        self.y = 0.0 # imaginary part
    
    def input(self, real, imaginary):
        self.x = real
        self.y = imaginary
    
    def show_complex_number(self):
        # print(f'{self.x} + {self.y}i')
        print(f"{self.x} + {self.y:.2f}i")

def add(c1, c2):
    result = ComplexNumber()
    result.x = c1.x + c2.x
    result.y = c1.y + c2.y
    return result

if __name__ == '__main__':
    A = ComplexNumber()
    B = ComplexNumber()
    C = ComplexNumber()

    A.input(3.1, 5.65)
    B.input(2.75, 1.2)

    C = add(A, B)
    print("A = ", end="")
    A.show_complex_number()
    print("B = ", end="")
    B.show_complex_number()
    print("C = ", end="")
    C.show_complex_number()